src/content/app/tools/blast/views/blast-form/BlastFormContext.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactNode, useState } from 'react';
+
+import useBlastForm from 'src/content/app/tools/blast/hooks/useBlastForm';
+
+type BlastFormContextType = {
+  updateSequenceValidityFlags: (index: number | null, status: boolean) => void;
+  removeSequenceValidityFlag: (index: number) => void;
+  clearSequenceValidityFlags: () => void;
+  sequenceValidityFlags: boolean[];
+};
+
+export const BlastFormContext = React.createContext<
+  BlastFormContextType | undefined
+>(undefined);
+
+type Props = {
+  children: ReactNode;
+};
+
+export const BlastFormContextContainer = (props: Props) => {
+  const [sequenceValidityFlags, setSequenceValidityFlags] = useState<boolean[]>(
+    []
+  );
+  const { sequences } = useBlastForm();
+
+  const updateSequenceValidityFlags = (
+    index: number | null,
+    status: boolean
+  ) => {
+    const array = [...sequenceValidityFlags];
+    index = index === null ? sequences.length : index;
+    array[index] = status;
+    setSequenceValidityFlags(array);
+  };
+
+  const removeSequenceValidityFlag = (index: number) => {
+    setSequenceValidityFlags(
+      sequenceValidityFlags.filter((value, i) => i !== index)
```

Comment:
So what happens when we are deleting the empty box that doesn't have an index?

Reply:
Yeah, that's what I thought — it's another way of breaking the UI:

https://user-images.githubusercontent.com/6834224/176670113-e465dc14-c9ae-456a-84b9-a00c4e25e375.mov

Reply:
Nothing happen :) And it wasn't that bad since it was getting overwritten the next time a new box is created but i agree its better to clean the array when this happen. I have updated the code to handle the deletion of empty box. But i spotted one more thing which is you can also delete a box by removing all the sequences in that box  and then click outside the box and the box will be deleted (this doesnt apply to the first box though). That behavior did surprise me but i am just trying to figure out the code where its deleting that box when its out of focus.